File: deal/src/main/java/ru/artemev/deal/client/ConveyorClient.java
package ru.artemev.deal.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import ru.artemev.deal.dto.CreditDTO;
import ru.artemev.deal.dto.LoanApplicationRequestDTO;
import ru.artemev.deal.dto.LoanOfferDTO;
import ru.artemev.deal.dto.ScoringDataDTO;

import java.util.List;

@FeignClient(name = "conveyor-client", url = "${conveyor.client.url}")
public interface ConveyorClient {
  @PostMapping("/conveyor/offers")
  List<LoanOfferDTO> getOffers(@RequestBody LoanApplicationRequestDTO loanApplicationRequestDTO);

  @PostMapping("/conveyor/calculation")
  CreditDTO getCreditDto(@RequestBody ScoringDataDTO scoringDataDTO);
}

File: deal/src/main/kotlin/ru/kit/deal/ConveyorClient.kt
package ru.kit.deal

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import ru.kit.deal.dto.LoanApplicationRequestDTO
import ru.kit.deal.dto.LoanOfferDTO

@FeignClient(name = "conveyor-client", url = "localhost:8080/api")
interface ConveyorClient {
    @PostMapping("/conveyor/offers")
    fun getOffers(@RequestBody loanApplicationRequestDTO: LoanApplicationRequestDTO): List<LoanOfferDTO>
}